src/auth/src/credentials.rs

```diff
@@ -165,3 +165,107 @@ pub mod traits {
         }
     }
 }
+
+/// The path to Application Default Credentials (ADC), as specified in [AIP-4110].
+///
+/// [AIP-4110]: https://google.aip.dev/auth/4110
+#[allow(dead_code)] // TODO(#442) - implementation in progress
+fn adc_path() -> Option<String> {
+    if let Ok(e) = std::env::var("GOOGLE_APPLICATION_CREDENTIALS") {
+        return Some(e);
+    }
+    adc_well_known_path()
+}
+
+/// The well-known path to ADC on Windows, as specified in [AIP-4113].
+///
+/// [AIP-4113]: https://google.aip.dev/auth/4113
+#[cfg(target_os = "windows")]
+fn adc_well_known_path() -> Option<String> {
+    if let Ok(root) = std::env::var("APPDATA") {
+        return Some(root + "/gcloud/application_default_credentials.json");
+    }
+    None
+}
+
+/// The well-known path to ADC on Linux and Mac, as specified in [AIP-4113].
+///
+/// [AIP-4113]: https://google.aip.dev/auth/4113
+#[cfg(not(target_os = "windows"))]
+fn adc_well_known_path() -> Option<String> {
+    if let Ok(root) = std::env::var("HOME") {
+        return Some(root + "/.config/gcloud/application_default_credentials.json");
+    }
+    None
```

Comment:
Not sure if this is better, but you could write:

```suggestion
    std::env::var("HOME").ok().map(|root| root + "/.config/gcloud/application_default_credentials.json")
```

Reply:
Another alternative is something like:

```rust
let root = std::env::var("HOME").ok()?;
Some(root + "/. blah blah")
```

The `?` thing works for `Result<>` and `Option<>` (I would say something about monads, but I left my smugness at home).

Reply:
Done.